File: tests/java/src/main/java/com/escape/tests/cases/Case217InterfaceSinkBridge02.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 217: interface_sink_bridge_02 deep stress pattern. */
public class Case217InterfaceSinkBridge02 {
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    private interface Sink {
        void put(Map<String, String> obj);
    }

    private static void sink(Map<String, String> obj) {
        RETAINED_AUDIT.add(obj);
    }

    public static String execute(String input) {
        String taskName = "interface_sink_bridge_02";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "extreme");
        payload.put("stage", "stress");
        payload.put("input", raw);
        payload.put("checksum", taskName + ":" + raw.length());
        Sink bridge = obj -> RETAINED_AUDIT.add(obj);
        // ESCAPE: interface bridge retains payload.
        bridge.put(payload);
        return "ok";
    }
}

File: tests/go/cases/case_217_interface_sink_bridge_02.go
package escape_tests

var retainedCase217 = []map[string]string{}

type case217Sinker interface{ Put(map[string]string) }
type case217Bridge struct{}

func (case217Bridge) Put(v map[string]string) {
	retainedCase217 = append(retainedCase217, v)
}

func Case217InterfaceSinkBridge02(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "interface_sink_bridge_02",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "interface_sink_bridge_02:" + raw,
	}
	var s case217Sinker = case217Bridge{}
	// ESCAPE: interface bridge dispatch to sink.
	s.Put(payload)
	return "ok"
}
